core/test-api/elasticsearch/src/main/java/org/opennms/core/test/elastic/ElasticSearchServerConfig.java

```diff
@@ -0,0 +1,124 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2017-2017 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2017 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.core.test.elastic;
+
+import java.util.Objects;
+
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.common.network.NetworkModule;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.node.Node;
+
+public class ElasticSearchServerConfig {
+
+    private static final String DEFAULT_HOME_DIRECTORY = "target/elasticsearch-home";
```

Comment:
Can we change the default behavior so that either A) a temporary directory is used, or B) this directory is cleared after a successful test so that multiple tests in the same module can use this rule without worrying the state of the previous test.

Reply:
I had some issues when doing that, as the Node is doing some magic when instantiating (even before the node is started) which caused some side-effects. I then decided to make it a non-dynamic directory. I can give it another try, though.